lib/TcpConnection.js

```diff
@@ -92,23 +97,17 @@ module.exports = function factory (userOptions) {
   }
 
   function reconnect () {
-    const retry = backoff.fibonacci()
-    retry.failAfter(options.reconnectTries)
-    retry.on('ready', () => {
-      connect((err) => {
-        if (connected === false) return retry.backoff(err)
-      })
-    })
-    retry.on('fail', (err) => process.stderr.write(`could not reconnect: ${err.message}`))
-    retry.backoff()
+    retryBackoff.backoff()
   }
   // end: connection handlers
 
   // begin: connection listeners
   function closeListener (hadError) {
     disconnect()
     if (hadError) {
-      process.stderr.write(socketError.message)
```

Comment:
Would you mind keeping this as the default onSocketClose implementation?

Reply:
Done!